neo/api/JSONRPC/JsonRpcApi.py

```diff
@@ -555,7 +555,11 @@ def process_transaction(self, contract_tx, fee=None, address_from=None, change_a
         self.wallet.Sign(context)
         if context.Completed:
             tx.scripts = context.GetScripts()
-            NodeLeader.Instance().Relay(tx)
-            return tx.ToJson()
+            relayed = NodeLeader.Instance().Relay(tx)
+            if relayed:
+                self.wallet.SaveTransaction(tx)
+                return tx.ToJson()
+            else:
+                raise JsonRpcError(-32001, "Could not relay tx %s " % tx.Hash.ToString())
```

Comment:
Unfortunately accepting this change would make the behaviour of calls like `sendtoaddress` deviate from the C# implementation, which is not what we want. See:
https://github.com/neo-project/neo/blob/dbe688b7b1d64975128b8aefb5c3ee692df07cd5/neo/Network/RPC/RpcServer.cs#L525-L532